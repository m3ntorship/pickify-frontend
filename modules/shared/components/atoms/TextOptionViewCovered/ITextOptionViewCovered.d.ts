import type React from 'react';

declare namespace ITextOptionViewCovered {
  export interface IProps extends IData, IHandlers {
    letter?: string;
  }
  export interface IData {
    id: string;
    optionBody: string;
  }
  export interface IHandlers {
    onOptionClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  }
}

export { ITextOptionViewCovered };
