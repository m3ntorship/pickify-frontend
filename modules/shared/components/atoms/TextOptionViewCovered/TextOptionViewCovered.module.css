.btnBody {
  @apply flex justify-between items-center rounded-md w-full focus:outline-none bg-grey-bg hover:bg-grey-shd6 pr-xsvv pl-xs py-xs;
}
.btnBody:disabled {
  @apply cursor-default disabled:opacity-100;
}
.letter {
  @apply bg-white text-dark-grey font-normal capitalize rounded-sm mr-3 w-l h-l flex justify-center items-center;
}
.option {
  @apply text-sm capitalize text-dark;
}
.flex-container {
  @apply flex flex-row justify-center items-center;
}
