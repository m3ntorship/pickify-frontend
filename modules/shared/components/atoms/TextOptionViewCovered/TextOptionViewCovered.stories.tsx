import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import TextOptionViewCovered from './TextOptionViewCovered';
import type { ITextOptionViewCovered } from './ITextOptionViewCovered';

export default {
  title: 'Atoms/TextOptionViewCovered',
  component: TextOptionViewCovered,
  argTypes: {},
};

const Template: Story<ITextOptionViewCovered.IProps> = (args): ReactElement => (
  <TextOptionViewCovered {...args} />
);
export const Default = Template.bind({});
Default.args = {
  optionBody: 'option 1',
  letter: 'A',
};
