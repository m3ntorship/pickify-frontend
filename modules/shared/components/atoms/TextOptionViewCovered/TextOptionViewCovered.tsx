import React from 'react';
import classNames from 'classnames';
import type { FC, ReactElement } from 'react';
import type { ITextOptionViewCovered } from './ITextOptionViewCovered';
import styles from './TextOptionViewCovered.module.css';

const TextOptionViewCovered: FC<ITextOptionViewCovered.IProps> = ({
  letter,
  optionBody,
  id,
  onOptionClick,
}): ReactElement => {
  const textPoll = classNames(styles.btnBody);

  return (
    <button
      type="button"
      id={id}
      data-testid={id}
      className={textPoll}
      onClick={onOptionClick}
      disabled={false}
    >
      <span className={styles['flex-container']}>
        {letter && <span className={styles.letter}>{letter}</span>}
        <p className={styles.option} dir="auto">
          {optionBody}
        </p>
        <span />
      </span>
    </button>
  );
};
export default TextOptionViewCovered;
