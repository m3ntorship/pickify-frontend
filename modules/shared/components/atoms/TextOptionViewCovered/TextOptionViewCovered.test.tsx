import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import TextOptionViewCovered from './TextOptionViewCovered';

describe('Snapshot testing for OptionViewCoverd component', () => {
  it('Testing TextOptionViewCovered with the necessary props', () => {
    const tree = renderer
      .create(
        <TextOptionViewCovered
          id="test1"
          optionBody="option 1"
          letter="A"
          onOptionClick={(): string => {
            return 'test';
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Behavioural tests for TextOptionViewCovered component ', () => {
  const one = 1;
  it('should render "option 1 " as option body', () => {
    render(
      <TextOptionViewCovered
        id="test1"
        optionBody="option 1"
        letter="A"
        onOptionClick={(): string => {
          return 'test';
        }}
      />,
    );
    const testBtn = screen.getByText('option 1');
    expect(testBtn).toBeInTheDocument();
  });
  it('the option should be clicked once', () => {
    const handleClick = jest.fn();
    render(
      <TextOptionViewCovered
        id="test1"
        optionBody="option 1"
        letter="A"
        onOptionClick={handleClick}
      />,
    );
    const testBtn = screen.getByTestId('test1');
    userEvent.click(testBtn);
    expect(handleClick).toHaveBeenCalledTimes(one);
  });

  it('should be active while its covered', () => {
    const handleClick = jest.fn();
    render(
      <TextOptionViewCovered
        id="test1"
        optionBody="option 1"
        letter="A"
        onOptionClick={handleClick}
      />,
    );

    const testBtn = screen.getByTestId('test1');
    userEvent.click(testBtn);
    expect(handleClick).toBeCalled();
    expect(testBtn).toBeEnabled();
  });
});
