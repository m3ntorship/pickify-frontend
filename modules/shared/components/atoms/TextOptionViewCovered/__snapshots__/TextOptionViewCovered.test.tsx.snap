// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for OptionViewCoverd component Testing TextOptionViewCovered with the necessary props 1`] = `
<button
  className="btnBody"
  data-testid="test1"
  disabled={false}
  id="test1"
  onClick={[Function]}
  type="button"
>
  <span
    className="flex-container"
  >
    <span
      className="letter"
    >
      A
    </span>
    <p
      className="option"
      dir="auto"
    >
      option 1
    </p>
    <span />
  </span>
</button>
`;
